File: leetcode-101.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def isSymmetric(self, root: TreeNode) -> bool:
        def dfs(p, q):
            if p is None and q is None:
                return True
            if p is None or q is None:
                return False
            return (p.val == q.val) and dfs(p.left, q.right) and dfs(p.right, q.left)
        return dfs(root, root)


if __name__ == "__main__":
    so = Solution()
    a = [1, 2]
    b = [2, 1]
    print(a==b)

File: leetcode/leetcode-101.cpp
/*
 *@File Name          :leetcode-101.cpp
 *@Author             :cc
 *@Version            :1.0.0
 *@Date               :2022/7/7 9:12
 *@Description        :
 *@Function List      :
 *@History            :
 */

struct TreeNode {
  int       val;
  TreeNode* left;
  TreeNode* right;
  TreeNode() : val(0), left(nullptr), right(nullptr) {}
  TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
  TreeNode(int x, TreeNode* left, TreeNode* right)
      : val(x), left(left), right(right) {}
};

class Solution {
 public:
  bool dfs(TreeNode* l, TreeNode* r) {
    if (l == nullptr && r == nullptr) return true;
    if (l == nullptr || r == nullptr) return false;

    if (l->val != r->val) return false;
    return dfs(l->left, r->right) && dfs(l->right, r->left);
  }
  bool isSymmetric(TreeNode* root) {
    if (root == nullptr) return true;
    return dfs(root->left, root->right);
  }
};